### PR TITLE
Document ring buffer for event stream integration

### LINKED ISSUE #1265: Document ring buffer for event stream integration
Page: https://docs.sensu.io/sensu-enterprise/3.4/integrations/event_stream/

To do: Add `ring_buffer_size` attribute

Reference: https://github.com/sensu/sensu-enterprise/pull/423

Notes:
- Integer, default: 1000
- Enabled by default?
- Set to 0 to disable?

### PR DESCRIPTION
**Pull request by apaskulin**
## Description

Adds the `ring_buffer_size` attribute to the docs.

## Motivation and Context

Closes #1265

### CODE REVIEW

**Comment by portertech** [comment]
When the Event Stream experiences a connection interruption, the ring buffer will begin discarding older data once it hits its limit. The ring buffer protects Sensu Enterprise, stopping the Event Stream from consuming too much memory (which would result in a OOM crash).

**Comment by portertech** [comment]
Once the Event Stream connection is re-established, the remaining data in the ring buffer will be sent.

File: content/sensu-enterprise/3.4/integrations/event_stream.md [hunk 1]
```diff
@@ -78,6 +79,14 @@ type                 | Boolean
 default              | `false`
 example              | {{< highlight shell >}}"filter_ok_keepalives": true{{< /highlight >}}
 
+ring_buffer_size     | 
+---------------------|------
+description          | By default, the event stream integration uses a data stream buffer. The `ring_buffer_size` attribute sets the limit for writes to the buffer. To disable the buffer, set to `0`.
```
Comment (apaskulin) [inline content/sensu-enterprise/3.4/integrations/event_stream.md]:
```suggestion
description          | By default, the event stream integration uses a data stream buffer to protect Sensu Enterprise from reaching an out-of-memory state. The `ring_buffer_size` attribute sets the limit for writes to the buffer. In the event of a connection interruption, the buffer begins discarding older data once it reaches the limit. Once the connection is re-established, the event stream sends the remaining data in the buffer. To disable the buffer, set `ring_buffer_size` to `0`.
```

File: content/sensu-enterprise/3.4/integrations/event_stream.md [hunk 2]
```diff
@@ -78,6 +79,14 @@ type                 | Boolean
 default              | `false`
 example              | {{< highlight shell >}}"filter_ok_keepalives": true{{< /highlight >}}
 
+ring_buffer_size     | 
+---------------------|------
+description          | By default, the event stream integration uses a data stream buffer to protect Sensu Enterprise from reaching an out-of-memory state. The `ring_buffer_size` attribute sets the limit for writes to the buffer. In the event of a connection interruption, the buffer begins discarding older data once it reaches the limit. Once the connection is re-established, the event stream sends the remaining data in the buffer. To disable the buffer, set `ring_buffer_size` to `0`.
+required             | false
+type                 | Integer
+default              | `1000`
+example              | {{< highlight shell >}}"ring_buffer_size": 500{{< /highlight >}}
```
Comment (apaskulin) [inline content/sensu-enterprise/3.4/integrations/event_stream.md]:
```suggestion
example              | {{< highlight shell >}}"ring_buffer_size": 1500{{< /highlight >}}
```

File: content/sensu-enterprise/3.4/integrations/event_stream.md [hunk 3]
```diff
@@ -36,7 +36,8 @@ event handler (integration).
     "host": "127.0.0.1",
     "port": 3000,
     "filter_metrics": true,
-    "filter_ok_keepalives": true
+    "filter_ok_keepalives": true,
+    "ring_buffer_size": 500
```
Comment (apaskulin) [inline content/sensu-enterprise/3.4/integrations/event_stream.md]:
```suggestion
    "ring_buffer_size": 1500
```

**Comment by portertech** [comment]
@apaskulin to disable the ring buffer, set it to `nil`. I do like the `0`, but it will require a change.

File: content/sensu-enterprise/3.4/integrations/event_stream.md [hunk 4]
```diff
@@ -78,6 +79,14 @@ type                 | Boolean
 default              | `false`
 example              | {{< highlight shell >}}"filter_ok_keepalives": true{{< /highlight >}}
 
+ring_buffer_size     | 
+---------------------|------
+description          | By default, the event stream integration uses a data stream buffer to protect Sensu Enterprise from reaching an out-of-memory state. The `ring_buffer_size` attribute sets the limit for writes to the buffer. In the event of a connection interruption, the buffer begins discarding older data once it reaches the limit. Once the connection is re-established, the event stream sends the remaining data in the buffer. To disable the buffer, set `ring_buffer_size` to `0`.
```
Comment (apaskulin) [inline content/sensu-enterprise/3.4/integrations/event_stream.md]:
```suggestion
description          | By default, the event stream integration uses a data stream buffer to protect Sensu Enterprise from reaching an out-of-memory state. The `ring_buffer_size` attribute sets the limit for writes to the buffer. In the event of a connection interruption, the buffer begins discarding older data once it reaches the limit. Once the connection is re-established, the event stream sends the remaining data in the buffer. To disable the buffer, set `ring_buffer_size` to `nil`.
```

**Review by portertech — APPROVED** [review]

**Pull request merged**